grpc/server/plugins/action/actionrpc_impl.h

```diff
@@ -17,28 +18,28 @@ class ActionRPCImpl final : public ActionRPC::Service
         dc = dc_obj;
     }
 
-    Status Arm(ServerContext *context, const ::google::protobuf::Empty *request,
+    Status Arm(ServerContext *context, const UUID* request,
                dronecorerpc::ActionResult *response) override
     {
-        const Action::Result action_result = dc->device().action().arm();
+        const Action::Result action_result = dc->device(request->uuid()).action().arm();
         response->set_result(static_cast<dronecorerpc::ActionResult::Result>(action_result));
         response->set_result_str(Action::result_str(action_result));
         return Status::OK;
     }
 
-    Status TakeOff(ServerContext *context, const ::google::protobuf::Empty *request,
+    Status TakeOff(ServerContext *context, const UUID* request,
                    dronecorerpc::ActionResult *response) override
     {
-        const Action::Result action_result = dc->device().action().takeoff();
+        const Action::Result action_result = dc->device(request->uuid()).action().takeoff();
         response->set_result(static_cast<dronecorerpc::ActionResult::Result>(action_result));
         response->set_result_str(Action::result_str(action_result));
         return Status::OK;
     }
 
-    Status Land(ServerContext *context, const ::google::protobuf::Empty *request,
+    Status Land(ServerContext *context, const UUID* request,
                 dronecorerpc::ActionResult *response) override
     {
-        const Action::Result action_result = dc->device().action().land();
+        const Action::Result action_result = dc->device(request->uuid()).action().land();
```

Comment:
I think we can give option to user to send NULL if they are handling single device. Then we need a NULL check here.